daemon/daemon.go

```diff
@@ -482,10 +496,10 @@ func NewDaemon(dp datapath.Datapath, iptablesManager rulesManager) (*Daemon, *en
 
 	// This needs to be done after the node addressing has been configured
 	// as the node address is required as suffix.
-	// well known identities have already been initialized above
+	// well known identities have already been initialized above.
 	// Ignore the channel returned by this function, as we want the global
 	// identity allocator to run asynchronously.
-	cache.InitIdentityAllocator(&d, k8s.CiliumClient(), nil)
+	d.identityAllocator.InitIdentityAllocator(k8s.CiliumClient(), nil)
```

Comment:
Future improvement todo (not really related to your PR but would be nice): It's weird to create the `identityAllocator` above then initialize it here, makes it easy to misuse it in-between. I realise the comment here is helpfully defining what else has to happen first, so it would be nice if we can find a way to fold the creation/initialization together earlier, while still satisfying this dependency here.

^ Should be addressed in a future PR, not this one.